File: Data Structures & Algorithms/count-number-of-islands/submission-3.cpp
class Solution {
private:
    void dfs(int i,int j,vector<vector<char>> &grid,int &ans){
        if(i<0 or j<0 or i>=grid.size() or j>=grid[0].size() or grid[i][j]=='0'){
            return;
        }
      
        grid[i][j]='0';
        dfs(i+1,j,grid,ans);
        dfs(i,j-1,grid,ans);
        dfs(i,j+1,grid,ans);
        dfs(i-1,j,grid,ans);
        return;
    }
public:
    int numIslands(vector<vector<char>>& grid) {
        int ans=0;
        for(int i=0;i<grid.size();i++){
            for (int j=0;j<grid[0].size();j++){
                if(grid[i][j]=='1'){
                    ans+=1;
                    dfs(i,j,grid,ans);
                }
            }
        } 
        return ans;
    }
};

File: Data Structures & Algorithms/generate-parentheses/submission-3.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        ans=[]
  
        def dfs(skobi,zbor,num):
            if(zbor<0):
                return 
            opened=(zbor+num)//2
            if(opened>n or opened-zbor>n):
                return
            if(num==2*n and zbor==0):
                ans.append(skobi)
            dfs(skobi+'(', zbor+1,num+1)
            dfs(skobi+')',zbor-1,num+1)
            return 
        dfs('',0,0)
        return ans